File: delivery-info-service/src/test/java/com/potatorider/deliveryinfoservice/domain/DeliveryTests.java
package com.potatorider.deliveryinfoservice.domain;

import static com.potatorider.domain.DeliveryStatus.ACCEPT;
import static com.potatorider.domain.DeliveryStatus.COMPLETE;
import static com.potatorider.domain.DeliveryStatus.REQUEST;
import static org.assertj.core.api.Assertions.assertThat;

import com.potatorider.domain.Delivery;
import org.junit.jupiter.api.Test;

public class DeliveryTests {

    @Test
    void next_status_is_not_last() {
        // Arrange
        Delivery delivery = DeliverySteps.makeValidDeliveryWithDeliveryStatus(REQUEST);

        // Act
        final Delivery response = delivery.nextStatus();

        // Assert
        assertThat(response.getDeliveryStatus()).isEqualTo(ACCEPT);
    }

    @Test
    void next_status_is_last() {
        // Arrange
        Delivery delivery = DeliverySteps.makeValidDeliveryWithDeliveryStatus(COMPLETE);

        // Act
        final Delivery response = delivery.nextStatus();

        // Assert
        assertThat(response.getDeliveryStatus()).isEqualTo(COMPLETE);
    }

}

File: delivery-info-service/src/test/java/com/potatorider/deliveryinfoservice/domain/DeliveryTests.kt
package com.potatorider.deliveryinfoservice.domain

import com.potatorider.domain.DeliveryStatus.*
import org.assertj.core.api.Assertions.*
import org.junit.jupiter.api.Test

class DeliveryTests {
    @Test
    fun next_status_is_not_last() {
        // Arrange
        val delivery = makeValidDeliveryWithDeliveryStatus(REQUEST)

        // Act
        val response = delivery.nextStatus()

        // Assert
        assertThat(response.deliveryStatus).isEqualTo(ACCEPT)
    }

    @Test
    fun next_status_is_last() {
        // Arrange
        val delivery = makeValidDeliveryWithDeliveryStatus(COMPLETE)

        // Act
        val response = delivery.nextStatus()

        // Assert
        assertThat(response.deliveryStatus).isEqualTo(COMPLETE)
    }
}
